wandb/sdk/wandb_run.py

```diff
@@ -1864,67 +1897,103 @@ def _save(
     ) -> Union[bool, List[str]]:
         if policy not in ("live", "end", "now"):
             raise ValueError(
-                'Only "live" "end" and "now" policies are currently supported.'
+                'Only "live", "end" and "now" policies are currently supported.'
             )
+
+        # Preserved for compatibility.
         if isinstance(glob_str, bytes):
             glob_str = glob_str.decode("utf-8")
         if not isinstance(glob_str, str):
             raise ValueError("Must call wandb.save(glob_str) with glob_str a str")
 
-        if base_path is None:
-            if os.path.isabs(glob_str):
-                base_path = os.path.dirname(glob_str)
-                wandb.termwarn(
-                    "Saving files without folders. If you want to preserve "
-                    "sub directories pass base_path to wandb.save, i.e. "
-                    'wandb.save("/mnt/folder/file.h5", base_path="/mnt")',
-                    repeat=False,
-                )
-            else:
-                base_path = "."
-        wandb_glob_str = GlobStr(os.path.relpath(glob_str, base_path))
-        if ".." + os.sep in wandb_glob_str:
-            raise ValueError("globs can't walk above base_path")
-
-        with telemetry.context(run=self) as tel:
-            tel.feature.save = True
-
+        # Provide a better error message to confused souls.
         if glob_str.startswith("gs://") or glob_str.startswith("s3://"):
             wandb.termlog(
                 "%s is a cloud storage url, can't save file to wandb." % glob_str
             )
             return []
-        files = glob.glob(os.path.join(self._settings.files_dir, wandb_glob_str))
-        warn = False
-        if len(files) == 0 and "*" in wandb_glob_str:
-            warn = True
+
+        if ".." + os.sep in glob_str:
+            raise ValueError("glob may not walk up directories using '..'")
```

Comment:
i think the intent was that your glob can go above the base_path, not that you can't use`..` at all if i read the code correctly

Reply:
ah, good catch!

Reply:
whew, that was a tough fix, considering interactions with "*" and other requirements. I rewrote this code to use `pathlib` and canonicalized inputs before calling `_save()` to simplify things.